src/rules.js

```diff
@@ -60,7 +60,7 @@ block._tag = 'address|article|aside|base|basefont|blockquote|body|caption'
   + '|legend|li|link|main|menu|menuitem|meta|nav|noframes|ol|optgroup|option'
   + '|p|param|section|source|summary|table|tbody|td|tfoot|th|thead|title|tr'
   + '|track|ul';
-block._comment = /<!--(?!-?>)[\s\S]*?-->/;
+block._comment = /<!--(?!-?>)[\s\S]*?--!?>/;
```

Comment:
see comment https://github.com/markedjs/marked/issues/1738#issuecomment-670055523

Reply:
```suggestion
block._comment = /<!--(?!-?>)[\s\S]*?(?:-->|$)/;
```

It looks like this would make marked work like CommonMark/GFM as shown in https://github.com/markedjs/marked/issues/1738#issuecomment-670138374